src/streaming/multiplex/mod.rs

```diff
@@ -32,7 +63,7 @@ pub struct StreamingMultiplex<B>(B);
 /// Additional transport details relevant to streaming, multiplexed protocols.
 ///
 /// All methods added in this trait have default implementations.
-pub trait Transport<ReadBody>: 'static +
+pub trait Transport<RID, ReadBody>: 'static +
```

Comment:
Again, didn't you want to use `RequestId`? Also, this seems to be inconsistent capitalization (the other one was `RId`).